cmip6_downscaling/methods/bcsd.py

```diff
@@ -316,63 +443,80 @@ def return_x_predict_rechunked(
         Date for prediction period end (e.g. '2090')
     variable: str
         The variable included in the dataset.
+    latmin : str
+            From run hyperparameters
+    latmax : str
+            From run hyperparameters
+    lonmin : str
+            From run hyperparameters
+    lonmax : str
+            From run hyperparameters
 
     Returns
     -------
     xr.Dataset
-        x prediction rechunked dataset
+        gcm predict rechunked dataset
     """
-    X_predict = load_cmip(
+    gcm_predict_ds = load_cmip(
         source_ids=gcm,
         activity_ids='ScenarioMIP',
         experiment_ids=scenario,
         variable_ids=[variable],
         return_type='xr',
-    ).sel(time=slice(predict_period_start, predict_period_end))
+    )
+
+    gcm_predict_ds_subset = subset_dataset(
+        gcm_predict_ds, predict_period_start, predict_period_end, latmin, latmax, lonmin, lonmax
+    )
+
     # validate that X_predict spatial chunks match those of X_train since the spatial chunks of predict data need
     # to match when they get passed to the fit_and_predict utility
     # if they are not, rechunk X_predict to match those spatial chunks specifically (don't just pass lat/lon as the chunking dims)
     matching_chunks_dict = {
         variable: {
-            'time': X_train_rechunked_ds.chunks['time'][0],
-            'lat': X_train_rechunked_ds.chunks['lat'][0],
-            'lon': X_train_rechunked_ds.chunks['lon'][0],
+            'time': gcm_train_subset_full_time_ds.chunks['time'][0],
+            'lat': gcm_train_subset_full_time_ds.chunks['lat'][0],
+            'lon': gcm_train_subset_full_time_ds.chunks['lon'][0],
         }
     }
-    x_predict_rechunked_ds, X_predict_rechunked_path = rechunk_zarr_array(
-        X_predict,
+    gcm_predict_rechunked_ds, _ = rechunk_zarr_array(
+        gcm_predict_ds_subset,
         zarr_array_location=None,
         variable=variable,
         chunk_dims=matching_chunks_dict,
         connection_string=connection_string,
         max_mem='1GB',
     )
-    return x_predict_rechunked_ds
+    return gcm_predict_rechunked_ds
 
 
 def fit_and_predict(
-    x_train_rechunked_ds: xr.Dataset,
-    y_rechunked_ds: xr.Dataset,
-    x_predict_rechunked_ds: xr.Dataset,
+    gcm_train_subset_full_time_ds: xr.Dataset,  # x_train_rechunked_ds: xr.Dataset,
```

Comment:
can remove the comments on the end here if we're doing away with the `x_train` and `Y_train` convention.